workspaces/arborist/lib/arborist/reify.js

```diff
@@ -12,6 +12,7 @@ const hgi = require('hosted-git-info')
 const rpj = require('read-package-json-fast')
 
 const { dirname, resolve, relative, join } = require('node:path')
+const { existsSync } = require("node:fs")
```

Comment:
We should use `exists` from the already imported `node:fs/promises` and await it.

Reply:
I would like to make this correction, however I am failing to locate a definition for `exists` in the `node:fs/promises` module.  
At first VScode intellisense was not showing any definitions for the given method (although it did find `exists` in `node:fs`), so I decided to double-check in NodeJS runtime. Below is the output;
```bat
C:\Users\User>node --version
v20.18.0

C:\Users\User>node
Welcome to Node.js v20.18.0.
Type ".help" for more information.
> require("node:fs/promises").exists
undefined
>
```

Reply:
Ah that's right, `exists` is deprecated.  `fs.stat` or `fs.access` is recommended instead, depending on the context.  `access` would be best here as we do not need the full stat info.

Reply:
I have made another commit, this time replacing `existsSync` with `access` from `node:fs/promises`.
In this commit, if `access` raises an "ENOENT" `Error`, the directory is created. Any other types of access errors that are raised, are allowed to bubble up.